cloudevents/spec.md

```diff
@@ -142,6 +145,10 @@ metadata typically allows for extension attributes. In other words, a binary
 formatted CloudEvent would work for both a CloudEvents enabled receiver as well
 as one that is unaware of CloudEvents.
 
+A "batch-mode message" is one where multiple (zero or more) events are
+encoded in a single message body, according to a specific event format. Not
+all event formats or protocol bindings support batch-mode messages.
```

Comment:
Possibly. I'm not sure. There's also (somewhere - I can't remember where) a prohibition on batches having multiple events with different spec versions. Which suggests there *is* a relationship between them, if only "they're all the same spec version".

Reply:
See here: https://github.com/cloudevents/spec/blob/main/cloudevents/bindings/http-protocol-binding.md#332-event-data-encoding

While we cleaning this up it might be good to be consistent and have this requirement be at the global level and not specific to http - for consistency.

re: no relationship... not sure requiring them to all be of the same CE version is the type of "relationship" people would think of, but I'm sure we could find the right wording. Unless you disagree with the idea of making sure people don't infer some semantic relationship...

Reply:
@jskeet to open-up a new PR to deal with this.